ratis-server/src/main/java/org/apache/ratis/server/storage/FileChunkReader.java

```diff
@@ -66,14 +79,23 @@ public FileChunkProto readFileChunk(int chunkMaxSize) throws IOException {
     final long remaining = info.getFileSize() - offset;
     final int chunkLength = remaining < chunkMaxSize ? (int) remaining : chunkMaxSize;
     final ByteString data = ByteString.readFrom(in, chunkLength);
+    boolean isDone = offset + chunkLength == info.getFileSize();
+    ByteString fileDigest = ByteString.EMPTY;
```

Comment:
Just set fileDigest to the current digest.
```
    final ByteString fileDigest = ByteString.copyFrom(
        digester != null? digester.digest(): info.getFileDigest().getDigest());
```

Reply:
done